torch/csrc/jit/codegen/cuda/ir_internal_nodes.h

```diff
@@ -513,61 +465,51 @@ class TORCH_CUDA_CU_API ReductionOp : public Expr {
 //! significant performance impact.
 class TORCH_CUDA_CU_API GroupedReductionOp : public Expr {
  public:
+  using Expr::Expr;
+
   GroupedReductionOp(
       IrBuilderPasskey,
-      std::vector<BinaryOpType> reduction_op_type,
+      std::vector<BinaryOpType> reduction_op_types,
       std::vector<Val*> init,
       std::vector<Val*> out,
       std::vector<Val*> in,
       bool is_allreduce = false);
 
-  GroupedReductionOp(const GroupedReductionOp* src, IrCloner* ir_cloner);
+  DECLARE_CLONE_AND_CREATE
 
   virtual const char* getOpString() const override {
     return "GroupedReductionOp";
   }
 
-  Expr* shallowCopy() const override;
-
   //! Number of expressions grouped horizontally. It does not reflect
   //! iteration grouping.
   size_t numExprs() const {
-    return reduction_op_types_.size();
+    return getReductionOpTypes().size();
   }
 
-  const std::vector<Val*>& initVals() const {
-    return init_vals_;
+  std::vector<Val*> initVals() const {
+    return {attributes().begin() + 2, attributes().end()};
```

Comment:
A bit concerning that we need to make sure we use correct offsets in the attribute vector. It seems like we are effectively losing static type safety, although the overall IR classes are more consistent.

Given that typically there are not many attributes in each class, I don't think the above concern is critical.